src/sinks/mod.rs

```diff
@@ -17,6 +17,8 @@ pub mod aws_s3;
 pub mod aws_sqs;
 #[cfg(feature = "sinks-azure_blob")]
 pub mod azure_blob;
+#[cfg(any(feature = "sinks-azure_blob", feature = "sinks-datadog_archives"))]
+pub mod azure_common;
```

Comment:
No need for the `sinks-datadog_archives` here, since you're including `sinks-azure_blob` in the flags set by `sinks-datadog_archive` already.